config.go

```diff
@@ -7,9 +7,15 @@ import (
 	"github.com/ghodss/yaml"
 )
 
+type Auth struct {
+	Enabled  bool   `json:"enabled"`
+	Audience string `json:"audience"`
+}
+
 type Endpoint struct {
-	Source  string `json:"source"`
-	KeyPath string `json:"keyPath"`
+	Source         string `json:"source"`
+	KeyPath        string `json:"keyPath"`
+	Authentication Auth   `json:"authentication"`
```

Comment:
I think the presence of an authentication value should be enough to signal that's it enabled, rather than having `Enabled` as field above. To do that, you can use a pointer value for this field, and check whether it's nil after decoding.